sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -344,7 +344,17 @@ private String ensureSlashPrefixed(String path) {
                        if (!(exception instanceof CosmosException)) {
                            // wrap in CosmosException
                            logger.error("Network failure", exception);
-                           dce = BridgeInternal.createCosmosException(request.requestContext.resourcePhysicalAddress, 0, exception);
+
+                           int statusCode = 0;
+                           if (WebExceptionUtility.isNetworkFailure(exception)) {
```

Comment:
I remember there was some retry policy code around 0 status code from GatewayStoreModel that we added last year.

@simplynaveen20 does this cause any issue in the ClientRetryPolicy or region fail over scneario?

Reply:
I changed the ClientRetryPolicy unit test to reflect that the status code would be != 0